### ISSUE TITLE
change in 3.8.3 breaks ruby 1.9

### ISSUE THREAD

**Issue by toy** [rspec/rspec-expectations#1113]
### Subject of the issue
Version 3.8.3 introduces a change which doesn't work in ruby 1.9.

### Your environment
* Ruby version: 1.9.3-p551
* rspec-expectations version: 3.8.3

### Steps to reproduce
```ruby
require 'delegate'
require 'pathname'
require 'rspec/autorun'

class A < DelegateClass(Pathname)
end

describe A.new('x') do
  it { should be_a(A) }
end
```

### Expected behavior
```
.

Finished in 0.00333 seconds (files took 0.06674 seconds to load)
1 example, 0 failures
```

### Actual behavior
```
F

Failures:

  1) x should be a kind of A
     Failure/Error: Unable to find -e to read failed line

     TypeError:
       bind argument must be an instance of Kernel
     # -e:1:in `block (2 levels) in <main>'

Finished in 0.00377 seconds (files took 0.11617 seconds to load)
1 example, 1 failure
```

**Comment by benoittgt** [comment]
Thanks for your report.

**Comment by benoittgt** [comment]
This PR will probably fix your issue https://github.com/rspec/rspec-expectations/pull/1112

**Comment by JonRowe** [comment]
@toy I believe this was fixed by #1112 if running against `master`/`3.8.4` doesn't fix this for you please re-open!

**Comment by toy** [comment]
@JonRowe Thanks, looks like 3.8.4 fixes the issue
